wallet/wallet.h

```diff
@@ -108,6 +108,9 @@ struct wallet_payment {
 	struct secret *path_secrets;
 	struct pubkey *route_nodes;
 	struct short_channel_id *route_channels;
+
+	/* The description of the payment. Must support `tal_len` */
+	char *description;
```

Comment:
This may be const?

This patch could have been folded into where the changes are used, but that's not worth rebasing for.